tests/sub_group_mask/sub_group_mask_common.h

```diff
@@ -0,0 +1,142 @@
+/*******************************************************************************
+//
+//  SYCL 2020 Extension Conformance Test
+//
+//  Provides common methods for sub_group_mask tests
+//
+*******************************************************************************/
+
+#ifndef __SYCLCTS_TESTS_SUB_GROUP_MASK_COMMON_H
+#define __SYCLCTS_TESTS_SUB_GROUP_MASK_COMMON_H
+
+#include "../common/common.h"
+#include "../common/type_coverage.h"
+
+#ifdef SYCL_EXT_ONEAPI_SUB_GROUP_MASK
+namespace {
+
+static const auto types =
+    named_type_pack<char, signed char, unsigned char, short, unsigned short,
+                    int, unsigned int, long, unsigned long, long long,
+                    unsigned long long>{
+        "char",           "signed char", "unsigned char",     "short",
+        "unsigned short", "int",         "unsigned int",      "long",
+        "unsigned long",  "long long",   "unsigned long long"};
+
+template <typename funT, typename PredT, typename T>
+class test_kernel;
+
+constexpr size_t globalSize = 128;
+constexpr size_t localSize = 32;
+
+struct even_predicate {
+  bool operator()(const sycl::sub_group &sub_group) {
+    return sub_group.get_local_id().get(0) % 2 == 0;
+  }
+};
+
+struct mod3_predicate {
+  bool operator()(const sycl::sub_group &sub_group) {
+    return sub_group.get_local_id().get(0) % 3 == 0;
+  }
+};
+
+struct true_predicate {
+  bool operator()(const sycl::sub_group &sub_group) { return true; }
+};
+
+struct false_predicate {
+  bool operator()(const sycl::sub_group &sub_group) { return false; }
+};
+
+struct first_half_predicate {
+  bool operator()(const sycl::sub_group &sub_group) {
+    return sub_group.get_local_id().get(0) <
+           sub_group.get_local_range().get(0) / 2;
+  }
+};
+
+struct second_half_predicate {
+  bool operator()(const sycl::sub_group &sub_group) {
+    return sub_group.get_local_id().get(0) >
+           sub_group.get_local_range().get(0) / 2 - 1;
+  }
+};
+
+inline auto get_result_array() {
+  std::array<bool, globalSize> resultArr;
+  resultArr.fill(true);
+  return resultArr;
+}
+
+inline bool if_check(const sycl::sub_group &sub_group) {
+#ifndef SYCL_CTS_ENABLE_FULL_CONFORMANCE
+  return sub_group.leader();
+#else
+  return true;
+#endif
```

Comment:
Avoiding negation makes code easier to read.
```suggestion
#ifdef SYCL_CTS_ENABLE_FULL_CONFORMANCE
  return true;
#else
  return sub_group.leader();
#endif
```